apps/extension/src/app/layout.tsx

```diff
@@ -1,17 +1,94 @@
+import { Button } from "@agreeto/ui";
+import { useEffect } from "react";
+
 import { Navbar } from "~app/navbar";
+import { useValidateTrialOrSub } from "~features/auth/is-authed";
+import { trpcApi } from "~features/trpc/api/hooks";
 
 export const Layout: React.FC<{ children?: React.ReactNode }> = ({
   children,
 }) => {
+  // Check that trial/subscription are valid
+  const renderComponent = useValidateTrialOrSub();
+
+  const utils = trpcApi.useContext();
+  useEffect(() => {
+    // prefetch subscription
+    utils.user.subscription.prefetch();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   return (
     <div className="flex w-full h-full">
       <aside className="w-1/12">
         <Navbar />
       </aside>
       {/* - 👇 inject the `children` here 👇 */}
       <main className="flex justify-center flex-1 w-11/12 overflow-y-hidden">
-        {children}
+        {renderComponent === "children" ? (
+          children
+        ) : renderComponent === "startTrial" ? (
+          <StartTrialScreen />
+        ) : renderComponent === "endTrial" ? (
+          <EndTrialScreen />
+        ) : renderComponent === "endSubscription" ? (
+          <EndSubscriptionScreen />
+        ) : null}
```

Comment:
Okay, so the `renderComponent` becomes the source of truth for which screens to display.

Since the routing really becomes relevant for the actual app, you pulled this out it seems like. 

It feels to me that our app has two different layouts:
- Full-Screen page (sign in)
- App Shell (navigation bar, calendar, sidebar)

In the app shell, we can disable the calendar + navigation and display an action sidebar. The action sidebar would contain the `StartTrialScreen`, `EndTrialScreen` & the `EndSubscriptionScreen`. 

The actual App Shell would contain the router (no routing in full screen mode anyways), so it seems to me that the only place to prefetch a query inside a component could be the component that decides between `SignIn` & `AppShell`.

In the AppShell, we could place all our fetching logic in the router -- no need to `useEffect`. 

The two mental models I see here:
1. fetch within Layouts (Nextjs-style)
2. fetch without Layouts (tanner-style)